src/app/clusters/thread_network_diagnostics_server/thread_network_diagnostics_server.cpp

```diff
@@ -59,15 +59,17 @@ CHIP_ERROR ThreadDiagosticsAttrAccess::Read(ClusterInfo & aClusterInfo, const At
 
     CHIP_ERROR err = ConnectivityMgr().WriteThreadNetworkDiagnosticAttributeToTlv(aClusterInfo.mFieldId, aEncoder);
 
-    *aDataRead = true;
-
-    // If it isn't a run time assigned attribute, e.j ClusterRevision, or if
-    // not implemented, use standard read.
-    // Clear error and no data read
+    // If it isn't a run time assigned attribute, e.g. ClusterRevision, or if
+    // not implemented, clear the error so we fall back to the standard read
+    // path.
+    //
+    // TODO: This is probably broken in practice.  The standard read path is not
```

Comment:
About this TODO, this check was for, on one hand, general attributes such as ClusterRevision that do give valid values with the standard read path, and, on a 2nd hand, for platforms that use NO_THREAD connectity manager implementation. Instead of erroring we would get default values as defined in the xml. 

For the 1st case, we could deal with those attribute IDs in _WriteThreadNetworkDiagnosticAttributeToTlv_ , set NO_ERROR and without trying to encode.
For the second case, we can flag an error i guess.

Reply:
If we use the no_thread connectivity manager implementation, arguably we should not be exposing this cluster at all.

But this does raise an interesting point: we may want to be able to do XML-defined defaults even if the attribute is backed by logic.  I'll think a bit more about how that should work.